src/variety/slither.js

```diff
@@ -294,13 +398,53 @@
 			"checkdir4BorderLine",
 
 			"checkOneLoop",
-			"checkDeadendLine+"
+			"checkDeadendLine+",
+
+			"checkSheepIn@swslither",
+			"checkWolvesOut@swslither"
 		],
 
 		checkdir4BorderLine: function() {
 			this.checkAllCell(function(cell) {
-				return cell.qnum >= 0 && cell.getdir4BorderLine1() !== cell.qnum;
+				return (
+					cell.qnum >= 0 &&
+					cell.qnum <= 4 &&
+					cell.getdir4BorderLine1() !== cell.qnum
+				);
 			}, "nmLineNe");
+		},
+
+		checkSheepIn: function() {
+			var bd = this.board;
+			if (!bd.scanInside()) {
+				return;
+			}
+			this.checkAllCell(function(cell) {
+				return cell.qnum === 5 && !cell.inside;
+			}, "nmOutside");
+		},
+
+		checkWolvesOut: function() {
+			var bd = this.board;
+			if (!bd.scanInside()) {
+				return;
+			}
+			this.checkAllCell(function(cell) {
+				return cell.qnum === 6 && cell.inside;
+			}, "nmInside");
+		}
+	},
+
+	"ImageTile@swslither": {
+		imgsrc_dataurl: "../img/sheep_wolf.png",
```

Comment:
I've tested this external file approach, and currently exporting SVG does not work. (within the editor, choose File > Save as image file). I think this is why Base64 is being used elesewhere in the codebase.

You can reduce the filesize of the image by running it through TinyPNG or pngcrush.

Reply:
I tested this too. It does work if you open the file with a viewer capable of retrieving web content, like a browser (as the image URL is embedded). I also tested it in node. It works there too but only when opening the file on the same system (as the file path is embedded).

I agree this isn't ideal ... but I also really didn't like the idea of embedding the base64 data for a few reasons. 1. the unwieldy size and/or the quality comprise. 2. the same image is used in multiple files, so the data would have to be copied, which makes maintenance awkward.

I thought about some other solutions. 1. Could update the grunt build process to embed the base64 at build time. 2. could modify the SVG export process to embed the base64 rather than the path (though not sure if this would require a change to pzpr-canvas, which I guess is not possible?).

Should I attempt one of these alternate solutions? or would you rather I just manually embed base64 in the 2 files?

Reply:
okay, so you want me to revert the KeyPopup to use emoji and embed a compressed version of the image into slither.js?

Reply:
Yes, you can return `KeyPopup` to the version in e4aa75aacd085d0eb95c823791829b9840114255 and use base64 in `slither.js`. I think the image can be compressed in several ways without a noticeable decline in quality.

Reply:
done